packages/plugin-ext/src/plugin/node/debug.ts

```diff
@@ -0,0 +1,346 @@
+            const languages = await adapterContribution.languages;
+            return languages || [];
+        }
+
+        return [];
+    }
+
+    async $getSchemaAttributes(contributionId: string): Promise<IJSONSchema[]> {
+        const adapterContribution = this.debugAdapterContributions.get(contributionId);
+        if (adapterContribution && adapterContribution.getSchemaAttributes) {
+            return adapterContribution.getSchemaAttributes();
+        }
+
+        return [];
+    }
+
+    async $getConfigurationSnippets(contributionId: string): Promise<IJSONSchemaSnippet[]> {
+        const adapterContribution = this.debugAdapterContributions.get(contributionId);
+        if (adapterContribution && adapterContribution.getConfigurationSnippets) {
+            return adapterContribution.getConfigurationSnippets();
+        }
+
+        return [];
+    }
+
+    async $provideDebugConfigurations(contributionId: string, folder: string | undefined): Promise<theia.DebugConfiguration[]> {
+        const adapterContribution = this.debugAdapterContributions.get(contributionId);
+        if (adapterContribution && adapterContribution.provideDebugConfigurations) {
+            const result = await adapterContribution.provideDebugConfigurations(undefined);
+            if (result) {
+                return result;
+            }
+        }
+
+        return [];
+    }
+
+    async $resolveDebugConfigurations(
+        contributionId: string,
+        debugConfiguration: theia.DebugConfiguration,
+        folder: string | undefined): Promise<theia.DebugConfiguration | undefined> {
+
+        const adapterContribution = this.debugAdapterContributions.get(contributionId);
+        if (adapterContribution && adapterContribution.resolveDebugConfiguration) {
+            return adapterContribution.resolveDebugConfiguration(debugConfiguration, folder);
+        }
+
+        return undefined;
+    }
+
+    protected async getExecutable(
+        packageContribution: PluginPackageDebuggersContribution | undefined,
+        adapterContribution: DebugAdapterContribution,
+        debugConfiguration: theia.DebugConfiguration): Promise<DebugAdapterExecutable> {
+
+        if (packageContribution && packageContribution.adapterExecutableCommand !== undefined) {
+            const result = await this.commandRegistryExt.executeCommand<DebugAdapterExecutable>(packageContribution.adapterExecutableCommand, []);
+            if (result) {
+                return result as DebugAdapterExecutable;
+            }
+        } else if (adapterContribution.provideDebugAdapterExecutable) {
+            const executable = await adapterContribution.provideDebugAdapterExecutable(debugConfiguration);
+            if (executable) {
+                return executable;
+            }
+        }
+
+        throw new Error('It is not possible to provide DebugAdapterExecutable');
+    }
+}
+
+class DebugPluginContribution extends VSCodeDebugAdapterContribution {
```

Comment:
it looks strange, plugins should not rely on from `VSCodeDebugAdapterContribution` debug extension, but should read metadata from plugins, the same way as metadata for commands and menu items and so on read by the plugin extension.